pkg/asset/installconfig/openstack/validation/machinepool.go

```diff
@@ -105,11 +98,23 @@ func validUUIDv4(s string) bool {
 	return true
 }
 
-func validateFlavor(flavor Flavor, req flavorRequirements, fldPath *field.Path) field.ErrorList {
+// validate flavor checks to make sure that a given flavor exists and meets the minimum requrement to run a cluster
+// this function does not validate proper install config usage
+// NOTE: Because we append the result of this function to an error list, it cannot be nil
+func validateFlavor(flavorName string, ci *CloudInfo, req flavorRequirements, fldPath *field.Path) field.ErrorList {
+	if flavorName == "" {
```

Comment:
could this possibly hide a problem?  The flavorName being passed int should never be empty. 
But here if it is, it is silently overlooked.  Maybe fill the errorList? or let it fall through 
The next lookup "ci.Flavors[flavorName]  will catch it

Reply:
That should be handled by the static usage validations. There are cases where it is valid for flavor name to be empty

Reply:
The purpose of this block of code is just to make sure that if a flavor is provided, that flavor is valid